### DISCUSSION TITLE
Placing Stop Loss on self.buy()

### DISCUSSION THREAD

**Discussion by ghost** [kernc/backtesting.py#299] [General]
Sorry for being daft but Im not tracking this in the documentation.

How do I set a trailing 80% stop loss when I initiate a trade? See the bold italics line below.

Really appreciate it.

```py
class SmaCross(Strategy):
    # Define the two MA lags as *class variables*
    # for later optimization
    n1 = 10
    n2 = 40
    
    def init(self):
        # Precompute the two moving averages
        self.sma1 = self.I(SMA, self.data.Close, self.n1)
        self.sma2 = self.I(SMA, self.data.Close, self.n2)
    
    def next(self):
        # If sma1 crosses above sma2, close any existing
        # short trades, and buy the asset
        if crossover(self.sma1, self.sma2):
            self.position.close()
            **_self.buy(sl=self.price*0.8)_**

        # Else, if sma1 crosses below sma2, close any existing
        # long trades, and sell the asset if you are long_short
        elif crossover(self.sma2, self.sma1):
            self.position.close()
 #           self.sell()
```

**Comment by kernc** [comment]
Your `sl=` above is initialized correctly, but it's not _trailing_ the price moves upwards.

[One of the examples](https://kernc.github.io/backtesting.py/doc/examples/Strategies%20Library.html) features [`TrailingStrategy`](https://kernc.github.io/backtesting.py/doc/backtesting/lib.html#backtesting.lib.TrailingStrategy) that does _roughly_ that. To use it, you need to transform your 20 % into [ATR](https://en.wikipedia.org/wiki/Average_true_range) units. Simplified:
```py
tr = data.High - data.Low
one_atr_in_pct = (tr / data.Close).mean()

def to_atr(pct):
    assert 0 < pct < 1, pct
    return pct / one_atr_in_pct

# >>> to_atr(.20)
# 12.214876329014
```

**Reply by ghost** [reply]
Thank you.
